contracts/router/modules/pool/velodrome/VelodromeV2Module.sol

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import {IERC20, SafeERC20} from "@openzeppelin/contracts-4.5.0/token/ERC20/utils/SafeERC20.sol";
+
+import {IndexedToken, IPoolModule} from "../../../interfaces/IPoolModule.sol";
+
+import {IVelodromeV2Pool} from "../../../interfaces/velodrome/IVelodromeV2Pool.sol";
+import {IVelodromeV2Router} from "../../../interfaces/velodrome/IVelodromeV2Router.sol";
+
+/// @notice PoolModule for Velodrome V2 pools
+/// @dev Implements IPoolModule interface to be used with pools added to LinkedPool router
+contract VelodromeV2Module is IPoolModule {
+    using SafeERC20 for IERC20;
+
+    IVelodromeV2Router public immutable router;
+
+    constructor(address _router) {
+        router = IVelodromeV2Router(_router);
+    }
+
+    function poolSwap(
+        address pool,
+        IndexedToken memory tokenFrom,
+        IndexedToken memory tokenTo,
+        uint256 amountIn
+    ) external returns (uint256 amountOut) {
+        // TODO: sanity checks on tokenFrom, tokenTo indices <= 1?
+        address tokenIn = tokenFrom.token;
+        IERC20(tokenIn).safeApprove(address(router), amountIn);
+
+        bool stable = IVelodromeV2Pool(pool).stable();
+        address factory = IVelodromeV2Pool(pool).factory();
+        IVelodromeV2Router.Route[] memory routes = new IVelodromeV2Router.Route[](1);
+        routes[0] = IVelodromeV2Router.Route({
+            from: tokenFrom.token,
+            to: tokenTo.token,
+            stable: stable,
+            factory: factory
+        });
+
+        uint256[] memory amounts = new uint256[](2);
+        amounts = router.swapExactTokensForTokens(amountIn, 0, routes, address(this), block.timestamp);
```

Comment:
Was the implementation where Module directly interacts with the pool (vs interacting with Velodrome Router) considered?

For instance, in #246 the Router interaction was chosen, as the direct interaction with UniV3 pool requires implementing a callback, which is not possible for a Module that gets delegate calls.

Reply:
Can do either way. Was being consistent with Uni v3 approach, but now I understand why you took the route you took. Will fix to go through pool to limit extra calls.